Dockerfile.md

```diff
@@ -1,47 +1,66 @@
 
-```
-docker run -t mytestssl --help
-docker run --rm -t mytestssl -p --header example.com
-```
+### From DockerHub or GHCR
 
-### From dockerhub or GHCR
+You can pull the image from either of these registries:
+- DockerHub: [`drwetter/testssl.sh`](https://hub.docker.com/r/drwetter/testssl.sh)
+- GHCR: [`ghcr.io/testssl/testssl.sh`](https://github.com/testssl/testssl.sh/pkgs/container/testssl.sh)
 
-You can pull the image from dockerhub or ghcr.io and run:
+Supported tags:
+- `3.2` / `latest`
+- `3.0` is the old stable version ([soon to become EOL](https://github.com/testssl/testssl.sh/tree/3.0#status))
 
-``docker run --rm -t drwetter/testssl.sh --fs example.com`` or ``docker run --rm -t ghcr.io/testssl/testssl.sh --fs example.com``
+### Building
 
-Supported tags are: ``3.2`` and ``latest``, which are the same. ``3.0`` is the old stable version which will be retired soon.
+You can build with a standard `git clone` + `docker build`. Tagging the image will make it easier to reference.
 
-``docker run --rm -t drwetter/testssl.sh:stable example.com`` or ``docker run --rm -t ghcr.io/testssl/testssl.sh:stable example.com``
+```bash
+mkdir /tmp/testssl && cd /tmp/testssl
+git clone --branch 3.2 --depth 1 https://github.com/testssl/testssl.sh .
+docker build --tag localhost/testssl.sh:3.2 .
+```
 
-Keep in mind that any output file (--log, --html, --json etc.) will be created within the container. If you wish to have this created in a local directory on your host you can mount a volume into the container and change the output prefix where the container user has write access to, e.g.:
+There are two base images available:
+- `Dockerfile` (openSUSE Leap), glibc-based + faster.
+- `Dockerfile-alpine` (Alpine), musl-based + half the size.
 
-```
-docker run --rm -t -v /tmp:/data drwetter/testssl.sh --htmlfile /data/ example.com
+Alpine is made available if you need broarder platform support or an image about 30MB smaller at the expense of speed.
+
+#### Remote build context + `Dockerfile`
+You can build with a single command instead via:
+
+```bash
+docker build --tag localhost/testssl.sh:3.2 https://github.com/testssl/testssl.sh.git#3.2
 ```
 
-which writes the HTML output to ``/tmp/example.com_p443-<date>-<time>.html.`` The uid/gid is the one from the docker user. Normally the file is 644. testssl.sh's docker container uses a non-root user (usually with user/groupid 1000:1000).
+This will produce a slightly larger image however as `.dockerignore` is not supported with remote build contexts.
+
+If you would like to build the Alpine image instead this way, just provide the alternative `Dockerfile` via `--file`:
+
+```bash
+docker build \
+  --tag localhost/testssl.sh:3.2-alpine \
+  --file https://raw.githubusercontent.com/testssl/testssl.sh/3.2/Dockerfile-alpine \
+  https://github.com/testssl/testssl.sh.git#3.2
+```
```

Comment:
Now that `Dockerfile.git` has become `Dockerfile-alpine`, perhaps the default `dist-git` should also be changed to `dist-local`?

This command presumably makes `dist-git` redundant, but I understand your reluctance to drop that from the 3.2 series and defer until 3.3.

As the file name has changed in the 3.2 series though, perhaps changing the default to `dist-local` is valid for consistency, and that would align better with these docs which don't mention using `--target` at all.

@jauderho may have an opinion on this as he contributed the `Dockerfile.git` file IIRC? Is this build command equivalent in functionality?

Reply:
As said, if you believe Dockerfile.alpine fits better, then feel free to change this. Maybe amend this PR accordingly or let me know if I should rather merge this first.